src/bin/pg_autoctl/fsm.c

```diff
@@ -228,13 +251,12 @@ KeeperFSMTransition KeeperFSM[] = {
 	{ SINGLE_STATE, WAIT_PRIMARY_STATE, COMMENT_SINGLE_TO_WAIT_PRIMARY, &fsm_prepare_replication },
 	{ PRIMARY_STATE, JOIN_PRIMARY_STATE, COMMENT_PRIMARY_TO_JOIN_PRIMARY, &fsm_prepare_replication },
 	{ PRIMARY_STATE, WAIT_PRIMARY_STATE, COMMENT_PRIMARY_TO_WAIT_PRIMARY, &fsm_disable_sync_rep },
-	{ STOP_REPLICATION_STATE, WAIT_PRIMARY_STATE, COMMENT_STOP_REPLICATION_TO_WAIT_PRIMARY, &fsm_promote_standby_to_primary },
 
 	/*
 	 * Situation is getting back to normal on the primary
 	 */
 	{ WAIT_PRIMARY_STATE, PRIMARY_STATE, COMMENT_WAIT_PRIMARY_TO_PRIMARY, &fsm_enable_sync_rep },
-	{ JOIN_PRIMARY_STATE, PRIMARY_STATE, COMMENT_JOIN_PRIMARY_TO_PRIMARY, NULL },
+	{ JOIN_PRIMARY_STATE, PRIMARY_STATE, COMMENT_JOIN_PRIMARY_TO_PRIMARY, &fsm_enable_sync_rep },
```

Comment:
This change seems to prevent failover if we have multiple stand-bys all with async replication (hence  num_sync_standbys == 0). The newly elected primary fails with:
```
10:31:09 33749 WARN  Failed to set the standby Target LSN because we don't have a quorum candidate yet
10:31:09 33749 ERROR Failed to transition from state "wait_primary" to state "primary", see above.
10:31:09 33749 ERROR Failed to transition to state "primary", retrying... 
```

What was the reason for adding this? I cannot remember

Reply:
That's important code when using sync rep. We want to wait until the standby nodes reached a recent enough LSN before considering the primary node ready for failover, and also this guarantees that the replication slot handling is not going to try to "advance" to values too far in the past, IIRC.

Now, in your test with only async nodes, it does not make sense to do that check. I have pushed some changes where we keep the `synchronousStandbyNames` in the `LocalPostgresServer` instance and then we can check it. When `number_sync_standbys` is zero, then `synchronousStandbyNames` is an empty string and we can bypass the extra checks.